src/RemoteInboxNotifications/StoredStateSetupForProducts.php

```diff
@@ -44,18 +44,18 @@ public static function init_stored_state( $stored_state ) {
 	 * @return bool
 	 */
 	private static function are_there_products() {
-		$product_query = new \WC_Product_Query(
-			array(
-				'limit'    => 1,
-				'paginate' => true,
-				'return'   => 'ids',
-				'status'   => array( 'publish' ),
-			)
+		global $wpdb;
+
+		$count = $wpdb->get_var(
+			"
+				SELECT COUNT(*)
+				FROM {$wpdb->posts}
+				WHERE post_type = 'product'
+				AND post_status = 'publish'
+			"
 		);
-		$products      = $product_query->get_products();
-		$product_count = $products->total;
 
-		return $product_count > 0;
+		return $count > 0;
```

Comment:
Why this change? I don't think we should be querying the posts table directly because the data store classes can be overridden (so the posts table may not be where products are stored).

Reply:
Could we lean on the product lookup table here possibly? https://github.com/woocommerce/woocommerce/blob/bf042a428d32442c6df923c888076935eea59cf5/includes/admin/class-wc-admin-dashboard.php#L231 - maybe a `count` of records there would do the trick

Reply:
The product lookup table includes _all_ products including trashed ones and there's no discriminator, so it would still need a join to the posts table.

Reply:
This same issue is blocking #4515 :-(

Reply:
Okay looking at some related code in Woo Core - https://github.com/woocommerce/woocommerce/blob/172ceb51e8e63f23142a71f23a2415bf1c5b938a/includes/class-wc-tracker.php#L333 - kind of _feels_ like this is inline with that logic there. @jeffstieler  definitely a valid question and concern but I think this query is pretty similar to the one we are using there for the tracker product count. And the same logic is powering the other note too so I feel its okay - especially given it is just powering the notes side of things.